catroid/src/org/catrobat/catroid/content/bricks/WhenConditionBrick.java

```diff
@@ -0,0 +1,165 @@
+			Drawable background = layout.getBackground();
+			background.setAlpha(alphaValue);
+
+			TextView whenLabel = (TextView) view.findViewById(R.id.when_conditon_label);
+			TextView whenLabel2 = (TextView) view.findViewById(R.id.when_condition_label_second_part);
+			TextView conditionEditText = (TextView) view.findViewById(R.id.brick_when_condition_edit_text);
+			whenLabel.setTextColor(whenLabel.getTextColors().withAlpha(alphaValue));
+			whenLabel2.setTextColor(whenLabel2.getTextColors().withAlpha(alphaValue));
+			conditionEditText.setTextColor(conditionEditText.getTextColors().withAlpha(alphaValue));
+			conditionEditText.getBackground().setAlpha(alphaValue);
+
+			this.alphaValue = alphaValue;
+		}
+
+		return view;
+	}
+
+	@Override
+	public View getView(Context context, int brickId, BaseAdapter baseAdapter) {
+		if (animationState) {
+			return view;
+		}
+		if (view == null) {
+			alphaValue = 255;
+		}
+
+		view = View.inflate(context, R.layout.brick_when_condition_true, null);
+		view = getViewWithAlpha(alphaValue);
+
+		setCheckboxView(R.id.brick_when_condition_checkbox);
+		final Brick brickInstance = this;
+
+		checkbox.setOnCheckedChangeListener(new CompoundButton.OnCheckedChangeListener() {
+			@Override
+			public void onCheckedChanged(CompoundButton buttonView, boolean isChecked) {
+				checked = isChecked;
+				adapter.handleCheck(brickInstance, isChecked);
+			}
+		});
+
+		TextView prototypeTextView = (TextView) view.findViewById(R.id.brick_when_condition_prototype_text_view);
+		TextView conditionEditText = (TextView) view.findViewById(R.id.brick_when_condition_edit_text);
+
+		getFormulaWithBrickField(BrickField.IF_CONDITION).setTextFieldId(R.id.brick_when_condition_edit_text);
+		getFormulaWithBrickField(BrickField.IF_CONDITION).refreshTextField(view);
+
+		prototypeTextView.setVisibility(View.GONE);
+		conditionEditText.setVisibility(View.VISIBLE);
+
+		conditionEditText.setOnClickListener(this);
+
+		return view;
+	}
+
+	public Formula getConditionFormula() {
+		return getFormulaWithBrickField(BrickField.IF_CONDITION);
+	}
+
+	@Override
+	public View getPrototypeView(Context context) {
+		View prototypeView = View.inflate(context, R.layout.brick_when_condition_true, null);
+		TextView textView = (TextView) prototypeView.findViewById(R.id.brick_when_condition_prototype_text_view);
+		textView.setText(String.valueOf(BrickValues.IF_CONDITION));
+		return prototypeView;
+	}
+
+	@Override
+	public List<SequenceAction> addActionToSequence(Sprite sprite, SequenceAction sequence) {
+		sequence.addAction(sprite.getActionFactory().createWaitUntilAction(sprite, getFormulaWithBrickField(BrickField.IF_CONDITION)));
+		return null;
+	}
+
+	@Override
+	public Script getScriptSafe() {
+		if (script == null) {
+			script = new WhenConditionScript(this);
+			formulaMap = script.getFormulaMap();
+		}
+		return script;
+	}
+
+	@Override
+	public Brick clone() {
+		// TODO clone script
+		return new WhenConditionBrick(script);
```

Comment:
Not absolutely sure, but I think it would be better to take null as parameter here and to override copyScriptForSprite(..) and set a reference/clone there (same as in the other ScriptBricks). 
Please check if copying of scripts/sprites, backpacking/unpacking works as expected